pkg/kubelet/metrics/metrics.go

```diff
@@ -822,6 +845,20 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		},
 	)
+
+	// ImagePullDuration is a Histogram that tracks the duration (in seconds) it takes for an image to be pulled,
+	// including the time spent in the waiting queue of image puller.
+	// The metric is broken down by bucketed image size.
+	ImagePullDuration = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           ImagePullDurationKey,
+			Help:           "",
+			Buckets:        podStartupDurationBuckets,
```

Comment:
why not setting the buckets here https://github.com/kubernetes/kubernetes/pull/121719/files#r1393461701?

I think you want two metrics: an histogram of image sizes and other of images download time

Reply:
The `Buckets` here is used to group the latencies. And the `imageSizeBuckets` are labels to be added to each latency datapoint to help us analyze the metric.

Two separate metrics will not help because we won't be able to associate the latency with the size.